### PR TITLE
Refactor Contact info interface

### PR DESCRIPTION
**Pull request by nicholaspcr**
#### Summary

Reference #5587 

#### Changes

- Separate `ValidateContactInfo` into: GetValidation, ValidateContactInfo(only acts on ContactInfo) and ExpireValidation.

#### Testing

unit tests

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/identityserver/bunstore/contact_info_store.go [hunk 1]
```diff
@@ -307,6 +277,110 @@ func (s *contactInfoStore) SetContactInfo(
 	return pbs, nil
 }
 
+func (s *contactInfoStore) ValidateContactInfo(ctx context.Context, v *ttnpb.ContactInfoValidation) error {
+	if len(v.GetContactInfo()) == 0 {
```
Comment (adriansmares) [inline pkg/identityserver/bunstore/contact_info_store.go]: Don't forget the span.

File: pkg/identityserver/bunstore/contact_info_store.go [hunk 2]
```diff
@@ -375,95 +449,45 @@ func (s *contactInfoStore) CreateValidation(
...
-func (s *contactInfoStore) Validate(ctx context.Context, validation *ttnpb.ContactInfoValidation) error {
-	ctx, span := tracer.StartFromContext(ctx, "Validate")
-	defer span.End()
-
-	// TODO: Refactor store interface to split this ito a separate methods.
-	// (https://github.com/TheThingsNetwork/lorawan-stack/issues/5587)
-
+func (s *contactInfoStore) GetValidation(
+	ctx context.Context, validation *ttnpb.ContactInfoValidation,
+) (*ttnpb.ContactInfoValidation, error) {
```
Comment (adriansmares) [inline pkg/identityserver/bunstore/contact_info_store.go:459]: Don't forget the span.

Comment (adriansmares) [inline pkg/identityserver/bunstore/contact_info_store.go:459]: Don't forget the span.

File: pkg/identityserver/bunstore/contact_info_store.go [hunk 3]
```diff
@@ -375,95 +449,45 @@ func (s *contactInfoStore) CreateValidation(
...
-
-	entityType, entityUUID, err := s.getEntity(store.WithSoftDeleted(ctx, false), entityID)
-	if err != nil {
-		return err
-	}
-
-	_, err = s.DB.NewDelete().
-		Model(&ContactInfo{}).
-		Where("entity_type = ? AND entity_id = ?", entityType, entityUUID).
+func (s *contactInfoStore) ExpireValidation(ctx context.Context, v *ttnpb.ContactInfoValidation) error {
```
Comment (adriansmares) [inline pkg/identityserver/bunstore/contact_info_store.go]: Don't forget the spans.

File: pkg/identityserver/bunstore/contact_info_store.go (hunk 1 above)
Comment (adriansmares) [inline pkg/identityserver/bunstore/contact_info_store.go]: To be honest `!=1` is what I would check here.

**Review by adriansmares — CHANGES_REQUESTED** [review]

File: pkg/identityserver/bunstore/contact_info_store.go [hunk 4]
```diff
@@ -375,95 +454,55 @@ func (s *contactInfoStore) CreateValidation(
...
-	return nil
+	return validationToPB(model), nil
 }
 
-func (s *contactInfoStore) DeleteEntityContactInfo(ctx context.Context, entityID ttnpb.IDStringer) error {
-	ctx, span := tracer.StartFromContext(ctx, "DeleteEntityContactInfo", trace.WithAttributes(
-		attribute.String("entity_type", entityID.EntityType()),
-		attribute.String("entity_id", entityID.IDString()),
+func (s *contactInfoStore) ExpireValidation(ctx context.Context, pb *ttnpb.ContactInfoValidation) error {
+	ctx, span := tracer.StartFromContext(ctx, "GetValidation", trace.WithAttributes(
```
Comment (adriansmares) [inline pkg/identityserver/bunstore/contact_info_store.go]: Wrong span name.

**Review by adriansmares — APPROVED** [review]

**Comment by nicholaspcr** [comment]
Re-requested the review by mistake @adriansmares

**Review by adriansmares — APPROVED** [review]

**Pull request merged**